src/include/util.h

```diff
@@ -323,6 +323,12 @@ namespace ranges
 		std::sort(std::begin(range), std::end(range), std::forward<Comparer>(comparer));
 	}
 
+	template <typename Range, typename Comparer = std::less<>>
+	bool is_sorted(Range &range, Comparer &&comparer = {})
```

Comment:
`const Range`.

Reply:
And if you are familiar enough, projection (see above `typename Proj = identity` for `find`)

So we would have:

```
bool is_BurningBuildingFramesSorted()
{
	return range::is_sorted(BurningBuildingFrames, std::greater<>{}, &Frame::Percent);
}
```

Reply:
I have seen projections, but is that possible with c++17 ?

Reply:
I mean like that:
```
template <typename Range, typename Comparer = std::less<>, typename Proj = identity>
bool is_sorted(const Range &range, Comparer &&comparer = {}, Proj &&proj = {}) {
    return std::is_sorted(std::begin(range), std::end(range),
                                    [&](const auto& lhs, const auto& rhs){
                                        return comparer(std::invoke(proj, lhs), std::invoke(proj, rhs);
                                    });
}
```

Reply:
That part I got rather quickly after I found the code above.

Reply:
done. Just needed to use a lambda for the projection (Otherwise the code would not be generic, or ugly)
e.g.
```
typedef typename std::iterator_traits<decltype(range.begin())>::value_type T;
...
then use &T::Percent

```

Reply:
Mostly because of intermediate `IsPercentDecreasingValid` (used only once).
If code is inlined/moved directly in `IsBurningBuildingFramesValid()`, it would be possible.

Reply:
Sure.
I still wonder if we should combine the percentage-lookup of construction and burning?
That would require the code to be generic/put in header.
How should the signature of such a percentage-lookup function look like?